Lab3/snake.js

```diff
@@ -727,19 +731,50 @@ function init() {
         document.getElementById("snake-name").setAttribute("placeholder", temp + "ss " + temp + "SSss--<");
     })
     // TODO: Set the Type field so that NO item is selected by default
-    document.getElementById("snake-type").selectedIndex = none;
+    document.getElementById("snake-type").selectedIndex = -1;
     // TODO: Any time the Name changes, automatically select the Type based on the first letter of the name
+    document.getElementById('snake-name').addEventListener('input', () => {
+        const name = settings.snakeName();
+        const typeCtrl = document.getElementById('snake-type');
+
+        if ( name === "" ) { return; }
+    
+        // Just look at the first character of the name and make the determination based on that
+        switch ( name[0] ) {
+            case 'S':
+                typeCtrl.selectedIndex = 0;
+                break;
+            case 'H':
+                typeCtrl.selectedIndex = 1;
+                break;
+            case 'T':
+                typeCtrl.selectedIndex = 2;
+                break;
+            case 'K':
+                typeCtrl.selectedIndex = 3;
+                break;
+            default:
+                typeCtrl.selectedIndex = -1;
+                break;
+        }
+    });
 
     // TODO: Validate the Name on every input to the Name field
     document.getElementById("snake-name").oninput = (event) => {
-        let characters = "HKSsT->";
-        if (!(characters.includes(event.date))) {
-            document.getElementById("snake-name").value = document.getElementById("snake-name").value.slice(0, document.getElementById("snake-name").value.length - 1)
+        const name = settings.snakeName();
+        
+        let strippedName = "";
+        for ( let c of name ) {
+            if ( "HKSsT -<".includes(c) ) {
+                strippedName += c;
+            }
         }
+
+        document.getElementById('snake-name').value = strippedName;
     }
     // TODO: Validate teh Name every time the Type field changes
-    document.getElemenyById("name").addEventListener("input", validateName());
-    document.getElementById("tpye").addEventListener("change", validateName());
+    document.getElementById("snake-name").addEventListener("input", validateName);
+    document.getElementById("snake-type").addEventListener("change", validateName);
```

Comment:
`addEventListener` is expecting a function object as its second parameter.  (It is a higher-order function!)  So you need to pass `validateName` as a function, NOT `validateName()` which is the value returned after calling the function